src/bindings/python/flux/uri/resolvers/lsf.py

```diff
@@ -0,0 +1,102 @@
+###############################################################
+# Copyright 2022 Lawrence Livermore National Security, LLC
+# (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+#
+# This file is part of the Flux resource manager framework.
+# For details, see https://github.com/flux-framework.
+#
+# SPDX-License-Identifier: LGPL-3.0
+###############################################################
+
+import re
+import subprocess
+from pathlib import PurePath
+
+from flux.uri import FluxURIResolver, URIResolverPlugin
+
+
+"""
+This module supports the Flux URI resolver for IBM Spectrum LSF.
+
+It can be used like:
+  `flux uri --remote lsf:<JOBID>`.
+
+"""
+
+
+def lsf_find_compute_node(jobid) -> str:
+    """Figure out where the job is being run using JSON output from `bjobs -l`."""
+
+    """
+    To get the name of the system (lassen or sierra or nid) without the node number,
+    pull the hostname and strip off anything that's not alphabetic.
+    """
+    host_system = subprocess.run(["hostname"], capture_output=True).stdout.decode(
+        "utf-8"
+    )
+    host_system = re.sub(r"[^a-zA-Z]", "", host_system)
+
+    """
+    A way more efficient way to do this could be revealed if we knew what API calls
+    LSF was making. There is API documentation in the LSF docs, but LSF is closed source.
+    """
+    sp = subprocess.run(["bjobs", "-l", str(jobid)], capture_output=True)
+    job_output = re.sub(r"\s", "", sp.stdout.decode("utf-8"))  ## removes whitespace
+    nodes = re.findall(
+        rf"[0-9]+\*{host_system}[0-9]+", job_output
+    )  ## returns list of possible nodes
+    """
+    Apply a logic to distinguish between batch nodes and launch nodes (that is, batch should be)
+    using more cores than launch regardless of center policies on this. Also, this logic should work at centers
+    that don't use launch nodes at all (unknown if OLCF does).
+    """
+    hostname, host_cores = "", 0
+    for node in nodes:
+        node = node.split("*")
+        if int(node[0]) > host_cores:
+            hostname = node[1]
+    if hostname != "":
+        return hostname
+    raise ValueError("Unable to resolve the hostname of the compute job.")
+
+
+def lsf_job_pid(jobid, hostname) -> str:
+    """Resolve the pids of an LSF job using ssh to the compute node"""
+    sp = subprocess.run(
+        ["ssh", hostname, "ps", "-Hu", "`whoami`", "|", "grep", "-m1", "flux-broker"],
```

Comment:
Ah. `ps -H` is used to make sure that we're getting the parent instance of any flux session running. From there if a user wants to connect to a child session, they can resolve that URI using the pid URI resolver. I'll check - maybe the command you've got would be compatible with `-H` too.

Reply:
Good point, I had forgotten that there might be a hierarchy of Flux instances. In that case, your use of `-H` here is clever.  ;-)

Also, I forgot that `ps` only prints processes in the current session (associated with the same terminal) so my suggestion won't work anyway. What you have here seems fine, though I wish there was a better way to select all processes for the calling user.

Actually, since the local and remote username must match (I assume this is required for LSF), then maybe the result of `os.getlogin()` could just be substituted, e.g.:

```suggestion
        ["ssh", hostname, "ps", "-Hu", os.getlogin(), "|", "grep", "-m1", "flux-broker"],
```

Reply:
> if a user wants to connect to a child session, they can resolve that URI using the pid URI resolver.

To clarify, once you have connected to a Flux instance, all sub-instances within that instance can most easily be resolved with the `jobid` resolver, which is more efficient, since the URI for jobs is stashed as a `memo` event in every job that is also an instance. Also, the `jobid` resolver is the default, so you can just use `flux uri JOBID` (or pass `JOBID` directly to any command that takes a high-level URI like `flux top`, `flux proxy`, `flux pstree`, etc) :slightly_smiling_face:

Reply:
Hmm, that's interesting. I'm getting the subprocess call to do all the options with: 
```python3
    sp = subprocess.run(
        ["ssh", hostname, "ps", "-Ho", "pid,comm", "-u", os.getlogin()],
        capture_output=True,
    )
```
returns:
```python3
(Pdb) print(sp.stdout.decode("utf-8"))
   PID COMMAND
 60425 sshd
 60426   ps
 52926 sshd
 52928   flux-broker-0
 53342     bash
 52828 jsmd
```

Reply:
Ah, sorry. I don't know why it was failing for me. User error probably!